docs/HowToPortTizenRTOnWiFiChipset.md

```diff
@@ -0,0 +1,127 @@
+# How to port TizenRT on new WiFi chipset
+
+There are two cases to consider towards porting TizenRT on a new WiFi chipset. The specific case depends on the vendor's decision to either use their proprietary supplicant codebase, or
+to directly use TizenRT's WiFi solution.
+
+Below are mentioned the two cases in detail. Additionally, instructions are provided to incorporate the WiFi chipset driver in TizenRT's code tree. Specifically, details are provided
+to interface driver to network stack, and to get/set WiFi configuration from a higher abstraction layer.
+ 
+## How to add external WiFi library to TizenRT
+
+External vendors may choose to build TizenRT application that includes their compiled WiFi libary, rather than opting for TizenRT's *wpa_supplicant*.
+To support external and independently compiled WiFi supplicant libraries, they must be merged with *libexternal.a* static libary that resides in the *external* folder of TizenRT.
+For details on how to merge libraries, please refer to [How to add Static Library](HowToAddStaticLibrary.md).
+
+
+
+## How to use WPA_SUPPLICANT
+If the developer wishes to use TizenRT's *wpa_supplicant* for WiFi-based applications, then following changes need to be performed at *[wpa_suppliant](#supplicant)*.
+
+### Supplicant
+
+Changes in *external* directory for a new WiFi board to include new chipset driver.
+
+1. Add new chipset driver under *external/wpa_supplicant/src/drivers/*, and name it as
+*driver_\<driver_name\>.c*
+
+2. Inside *driver_\<driver_name\>.c*, declare a driver structure variable named *wpa_driver_\<driver_name\>_ops*.
+*wpa_driver_\<driver_name\>_ops* is a structure of type *wpa_driver_ops* that is declared in *external/wpa_supplicant/src/drivers/driver.h*.
+The *wpa_driver_ops* structure contains function pointers to the specific WiFi driver that the supplicant links to.
+
+
+3. As an example, refer to the *wpa_driver_t20_ops* structure declared in 
+*external/wpa_supplicant/src/drivers/driver_t20.c* for the LSI WiFi chipset.
+```
+const struct wpa_driver_ops wpa_driver_t20_ops = {
+	.name = "slsi_t20",
+	.desc = "SLSI T20 Driver",
+	.init2 = slsi_t20_init,
+	.deinit = slsi_t20_deinit,
+	.get_mac_addr = slsi_get_mac_addr,
+	.get_capa = slsi_t20_get_capa,
+	.scan2 = slsi_hw_scan,…
+} 
+```
+
+4. Add the driver structure variable created in step 2 to the *wpa_drivers[]* list in *external/wpa_supplicant/src/drivers/drivers.c*, as shown below:
+```
+#ifdef CONFIG_DRIVER_<DRIVER_NAME>
+extern struct wpa_driver_ops wpa_driver_<driver_name>_ops;	/* driver_<driver_name>.c */
+#endif
+```
+This links the supplicant to the relevant WiFi driver.
+
+
+5. Make sure you have defined *CONFIG_DRIVER_\<DRIVER_NAME\>* using *external/wpa_supplicant/Make.defs*, as shown below:
```

Comment:
1. Path is wrong.
```
*external/wpa_supplicant/wpa_supplicant/Make.defs*
```
2. If we support one board, this is ok. But if we support several boards on wpa_supplicant, this is a wrong way.
It should be selected by architecture, not fixed value.